core/domain/role_services.py

```diff
@@ -0,0 +1,116 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module contains the Hierarchy Structure of roles for action
+inheritance, Actions permitted to the roles and the functions used to
+access roles and actions.
```

Comment:
This module contains the Hierarchy Structure of roles for 
action inheritance, permitted actions and the functions used to access them respectively.